### DISCUSSION TITLE
Stepwise forward addition process

### DISCUSSION THREAD

**Discussion by Percival24** [nlmixr2/nlmixr2#125] [Q&A] [answered]
Hello :) 
I learned that the objective function value could only decrease during the stepwise forward addition process. No matter what covariate I add, that value should always fall. A minimum decrease of at least 3.84 (alpha=0.05) is considered significant. So... From that theoretical point of view. I have played with the theophylline data and it turns out that my theoretical background and the output don't tell the same story :/  

Here is my example. 

`rm(list = ls())

library(ggplot2)
library(nlmixr2)
library("haven")
library("stringr")
library("ggplot2")
library(writexl)
library(xpose.nlmixr2)

theo_sd$lnWT = log(theo_sd$WT/mean(theo_sd$WT))

one.compartment <- function() {
  ini({
    tka <- 0.45 # Log Ka
    tcl <- 1 # Log Cl
    tv <- 3.45    # Log V
    eta.ka ~ 0.6
    [eta.cl](http://eta.cl/) ~ 0.3
    eta.v ~ 0.1
    [add.sd](http://add.sd/) <- 0.7
  })

  model({
    ka <- exp(tka + eta.ka)
    cl <- exp(tcl + [eta.cl](http://eta.cl/))
    v <- exp(tv + eta.v)
    d/dt(depot) = -ka * depot
    d/dt(center) = ka * depot - cl / v * center
    cp = center / v
    cp ~ add([add.sd](http://add.sd/))
  })
}

fit <- nlmixr2(one.compartment,theo_sd,est="focei",foceiControl(outerOpt="nlminb",print=0),table=tableControl(cwres=T, npde=T))

fit_Wt_Cl <- fit %>%
  model( { cl  <- exp(tcl + b1*lnWT + eta.clnm) } ) %>%
  ini (tka <- 0.45) %>%
  ini (tcl <- 1) %>%
  ini (tv <- 3.45)  %>%
  ini ([add.sd](http://add.sd/) <- 0.7) %>%
  ini (b1 = -1 ) %>%
  nlmixr2(theo_sd,est="focei",
          foceiControl(outerOpt="nlminb",print=0),
          table=tableControl(cwres=T, npde=T))

fit$objDf       # 116.8167
fit_Wt_Cl$objDf # 120.9443`

**Comment by Percival24** [comment, marked as answer]
Besides, during the backward elimination process, on my personal project, the value of the objective function decreases when I remove some covariates. This is as well theoretically forbidden :/

**Reply by mattfidler** [reply]
This theoretical background is not correct.  The objective function depends on the etas selected.  If you start at zero then you may have increases or decreases.  This is the same as seen in other modeling tools.

I typically look at the eta plots before committing to adding these to a covariate search.  If they do not appear to be significant they often are not significant.

**Reply by Percival24** [reply]
So even in the backward elimination steps, the objective function could decrease? 
What to do in such a scenario?  
Here is what happened to me for the first elimination step : 

Base model + Covariate_1 on Cl + Covariate_2 on Qm + Covariate_3 on Cl ....(obj: 31 337.06)  

Base model + Covariate_2 on Qm + Covariate_3 on Cl.......................................(obj: 31 313.44)                                                
Base model + Covariate_1 on Cl + Covariate_2 on Qm.......................................(obj: 31 325.66)                
Base model + Covariate_1 on Cl + Covariate_3 on Cl..........................................(obj: 31 344.52)

All the 3 models produce good fits and there is nothing wrong with their estimated parameters.

**Reply by mattfidler** [reply]
It could increase or decrease.  If it is well behaved it will do what you expect it to.  I have seen both with NONMEM and see no reason why it couldn't happen with `nlmixr2`.

For me, I typically look at:

- What is thought to be clinically relevant 
- What are the trends with the `etas`.

**Reply by Percival24** [reply]
Thanks, @mattfidler !!!!!! Very helpful once again :)
